src/drivers/xmc.c

```diff
@@ -0,0 +1,208 @@
+#include "net_builtin.h"
+
+#if MG_ENABLE_TCPIP && defined(MG_ENABLE_DRIVER_XMC) && MG_ENABLE_DRIVER_XMC
+
+#ifdef XMC4400_F100x512
+#include <XMC4400.h>
+#else
+#ifdef XMC4700_F144x2048
+#include <XMC4700.h>
+#endif
+#endif
```

Comment:
I decided to include cmsis, because defining the eth structure will look very ugly in this case.. Compared to other drivers we have, it has a lot bigger and when I saw how many lines it wasted (almost double than stm or imxrt), I decided to rather include cmsis header.. Plus, it helps a lot with code readability if we use the bit masks defined in the cmsis header, rather than MG_BIT(bit)..